46.md

```diff
@@ -2,161 +2,83 @@ NIP-46
 ======
 
 Nostr Connect
-------------------------
+-------------
 
 `draft` `optional`
 
-## Rationale
+This NIP describes a method for 2-way communication between a **remote signer** and a normal Nostr client. The remote signer could be, for example, a hardware device dedicated to signing Nostr events, while the client is a normal Nostr client.
 
-Private keys should be exposed to as few systems - apps, operating systems, devices - as possible as each system adds to the attack surface.
+## Signer Discovery
 
-Entering private keys can also be annoying and requires exposing them to even more systems such as the operating system's clipboard that might be monitored by malicious apps.
+The client must somehow be able to contact the signer through a common relay.
 
+### Started by the signer
 
-## Terms
-
-* **App**: Nostr app on any platform that *requires* to act on behalf of a nostr account.
-* **Signer**: Nostr app that holds the private key of a nostr account and *can sign* on its behalf.
+The remote signer generates a connection token in the form
 
+```
+<npub1...>#<secret>?relay=wss://...&relay=wss://...
+```
 
-## `TL;DR`
+The user copies that token and pastes it in the client UI somehow. Then the client can send events of kind `24133` to the specified relays and wait for responses from the remote signer.
```

Comment:
I think the term `<remote-pubkey>` and `<local-pubkey>` are better because the local/remote are reversed depending on which side its initiated from, and ofc `<remote-pubkey>` is always who you're supposed to `p` tag which wasn't clear in this rewrite

Reply:
only the client can initiate a request and only a signer can send a response. I don't think we need to worry about reverse situations..? maybe i'm not understanding what you are getting at.

right now the `p` tag seems really clear and explicit to me, the remote signer should tag the client pubkey, and the client should tag the remote signer pubkey.